htdocs/product/class/product.class.php

```diff
@@ -562,7 +568,11 @@ public function create($user, $notrigger = 0)
 		$error = 0;
 
 		// Clean parameters
-		$this->ref = dol_sanitizeFileName(dol_string_nospecial(trim($this->ref)));
+		if (!empty($conf->global->MAIN_DO_NOT_SANITIZE_PRODUCT_REF)) {
```

Comment:
Removing sanitization of ref means you can get a / into it. Ref is also used to be the directory name where files are stired. This will break a lot of features.
If you need to store a non sanitized ref to store a value with dangerous characters, you should add an complementary fields called ref_not_sanitized to store it for example.

Reply:
> Removing sanitization of ref means you can get a / into it.

En effet, nous avons beaucoup, beaucoup de ref de produit avec / depuis plus qu'un an.  Il y a plusieurs points :

- Nous avons importer tous ces / lorsqu'on a basculé via des scripts SQL à Dolibarr en production sans observer des problèmes d'utilisation.
- Nous avons enlevé sanitization pour des nouvelles créations de produit il y a an sans observer des problèmes.
- Nous avons examiné le code pour des utilisations des ref de produit et nous observons que les programmeurs utilisent les fonctions dol_sanitizeFileName ou sanitizeVal  lorsqu'ils obtiennent un ref de produit de la base pour des noms de fichiers ou dossiers.

> This will break a lot of features.

Apparemment non. Sérieusement, nous avons cherché et nous n'avons trouvé aucun problème.

En ce qui concerne la suggestion MAIN_SECURITY_ALLOW_UNSECURED_REF_LABELS, j'avoue qu'il faut que je réfléchisse avant que je puisse répondre.

Reply:
Bonjour,

Avec commit [1e17c43](https://github.com/marcusdeangabriel/no-sanitize-prod-ref/commit/1e17c43e992f66585f11a710d4f03e200234ad9d), j'ai pris en compte votre conseil d'utiilser MAIN_SECURITY_ALLOW_UNSECURED_REF_LABELS au lieu de MAIN_DO_NOT_SANITIZE_PRODUCT_REF.

J'ai corrigé la documentation de la fonction `GETPOST` dans `functions.lib.php`.  `nohtml` laisse passer `"` et `../` mais il était documenté comme `alphanohtml` qui enlève `"` et `../` ce qui m'a dérouté.

Reply:
I was preparing newer pull request than this current and and commited [7acef58](https://github.com/marcusdeangabriel/dolibarr/commit/7acef585ff2da7ca272516e4ab91326dbd9d39f2), which provoked the commit [3c556f9](https://github.com/Dolibarr/dolibarr/pull/23587/commits/3c556f9fc060bbf320cbe3541a5c5f426b72bb4d), by mistake.  These two commits should not be included for the current pull request.